estate/models/estate_property.py

```diff
@@ -0,0 +1,110 @@
+            ("offer_accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("canceled", "Canceled"),
+        ],
+        default="new",
+        required=True,
+        copy=False,
+        string="Status"
+    )
+
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(
+        "Available From",
+        copy=False, default=fields.Datetime.add(fields.Datetime.today(), months=3)
+    )
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer("Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        selection=[
+            ("north", "North"),
+            ("south", "South"),
+            ("east", "East"),
+            ("west", "West"),
+        ]
+    )
+    
+    property_type_id = fields.Many2one("estate.property.type",string="Property Type")
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    salesman_id = fields.Many2one("res.users", string="Salesman", default=lambda self: self.env.user)
+    tag_ids = fields.Many2many("estate.property.tag", string="Tag")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    
+    total_area = fields.Float("Total Area (sqm)", compute="_compute_area")
+    best_price = fields.Float("Best Offer", compute="_compute_best_price")
+    
+    @api.depends("living_area", "garden_area")
+    def _compute_area(self) -> None:
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+             
+    @api.depends("offer_ids")
+    def _compute_best_price(self) -> None:
+        for property in self:
+            property.best_price = max(property.offer_ids.mapped('price')) if property.offer_ids else 0
+            
+    @api.constrains("selling_price")
+    def _check_selling_price(self) -> None:
+        # Filter based on selling price because when we cancel a approved offer, the price is reset to 0
+        filtered_property = self.filtered("selling_price")
+        if any(tools.float_compare(p.selling_price, p.expected_price * 0.9, 0) == -1 for p in filtered_property):
+            raise ValidationError("Selling price to low (< 90% of expected price)")
+
+    @api.onchange("garden")
+    def _onchange_garden(self) -> None:
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = None
+            
+    @api.ondelete(at_uninstall=False)
+    def ondelete(self) -> None:
+        if any(p.state not in ("new", "canceled") for p in self):
+            raise UserError("State must be new or canceled")
```

Comment:
Not very clear for the user why he can't uninstall the app. Is the app state supposed to be in new?